benchmark/utils/timer_impl.hpp

```diff
@@ -87,13 +92,31 @@ class Timer {
     std::int64_t get_num_repetitions() const { return duration_sec_.size(); }
 
     /**
-     * Compute the average time of repetitions in seconds
+     * Compute the time from the given statistical method
+     *
+     * @param method  the statistical method
      *
-     * @return the average time in seconds
+     * @return the statistical time
      */
-    double compute_average_time() const
+    double compute_time(const std::string& method = "average") const
     {
-        return this->get_total_time() / this->get_num_repetitions();
+        if (method == "average") {
+            return this->get_total_time() / this->get_num_repetitions();
+        }
+        auto copy = duration_sec_;
+        std::sort(copy.begin(), copy.end());
+        if (method == "best") {
+            return copy.front();
+        } else if (method == "worst") {
+            return copy.back();
```

Comment:
how about using min and max? For times, this should be unambiguous and is closer to what is used in statistical analysis